packages/integrations/components/coral/usePicoObserver.js

```diff
@@ -0,0 +1,138 @@
+import { useState, useCallback, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import debounce from 'lodash/debounce';
+import {
+  actionReceiveCoralLogoutRequest,
+} from '../../actions/coralActions';
+import {
+  actionVerifyPicoUser,
+  actionRequireUpgrade,
+  actionReceivePicoPlanUpgrade,
+} from '../../actions/picoActions';
+import { showUpgradeModalSelector } from '../../selectors/coralSelector';
+
+/**
+ * A hook that mounts a MutationObserver on the global `PicoSignal-container`
+ * DOM node on the `pico.loaded` event. The observer dispatches actions used
+ * to verify that user with Pico and perform conditional rendering/SSO-token
+ * retrieval given the values of the Signal node's data attributes.
+ *
+ * @param {array} tiers Array of valid subscription tier IDs for Coral SSO.
+ */
+export default function usePicoObserver(tiers) {
```

Comment:
I need to do a whole round of clean up on this hook to make sure everything is working exactly as it should and will be writing unit tests for it in a future PR. This part of the integration is still in progress, but should not block the review of this work.